docs/plugin/custom-executor.md

```diff
@@ -0,0 +1,212 @@
+
+:::note
+Instead of GoReleaser, you can use another tool of your choice. The important thing is to produce the binaries for the architecture of the host platform where Botkube is running.
+:::
+
+1. Create the GoReleaser configuration file:
+
+   ```yaml
+   cat << EOF > .goreleaser.yaml
+   before:
+     hooks:
+       - go mod download
+
+   builds:
+     - id: echo
+       main: cmd/echo/main.go
+       binary: executor_echo_{{ .Os }}_{{ .Arch }}
+
+       no_unique_dist_dir: true
+       env:
+         - CGO_ENABLED=0
+       goos:
+         - linux
+         - darwin
+       goarch:
+         - amd64
+         - arm64
+       goarm:
+         - 7
+
+   snapshot:
+     name_template: 'v{{ .Version }}'
+   EOF
+   ```
+
+2. Build the binaries:
+
+   ```bash
+   goreleaser build --rm-dist --snapshot
+   ```
+
+Congrats! You just created your first Botkube executor plugin! :tada:
+
+Now it's time to [test it locally with Botkube](local-testing.md). Once you're done testing, see how to [distribute it](repository.md).
+
+## Passing configuration to your plugin
+
+Sometimes your executor plugin requires a configuration specified by the end-user. Botkube supports such requirement and provides an option to specify plugin configuration under `config`. An example Botkube configuration looks like this:
+
+```yaml
+communications:
+  "default-group":
+    slack:
+      channels:
+        "default":
+          name: "all-teams"
+          bindings:
+            executors:
+              - echo-team-a
+              - echo-team-b
+
+executors:
+  "echo-team-a": # executor configuration group name
+    botkube/echo:
+      enabled: true
+      config:
+        changeResponseToUpperCase: false
+  "echo-team-b": # executor configuration group name
+    botkube/echo:
+      enabled: true
+      config:
+        changeResponseToUpperCase: true
+```
+
+This means that two different `botkube/echo` plugin configurations were bound to the `all-teams` Slack channel. Under `executor.ExecuteInput{}.Configs`, you will find the list of configurations in the YAML format as specified under the `config` property for each bound and enabled executor. The order of the configuration is the same as specified under the `bindings.executors` property. It's up to the plugin author to merge the passed configurations. For example:
+
+```go
+// Config holds the executor configuration.
+type Config struct {
+	ChangeResponseToUpperCase *bool `yaml:"changeResponseToUpperCase,omitempty"`
+}
+
+func (EchoExecutor) Execute(_ context.Context, in executor.ExecuteInput) (executor.ExecuteOutput, error) {
+	finalCfg := Config{}
+	for _, inputCfg := range in.Configs {
+		var cfg Config
+		err := yaml.Unmarshal(inputCfg.RawYAML, &cfg)
+		if err != nil {
+			return executor.ExecuteOutput{}, err
+		}
+		if cfg.ChangeResponseToUpperCase == nil {
+			continue
+		}
+		finalCfg.ChangeResponseToUpperCase = cfg.ChangeResponseToUpperCase
```

Comment:
So this part confused me a bit, you accept multiple configs, iterate them and override it all the time until you get final config?

Reply:
yep, as we have multiple configs, and we need to merge them together. The `ChangeResponseToUpperCase` is a bool so the only way is to override it. Maybe I will change the example and use `slice` so I would be able to aggregate some settings together.

Reply:
I updated it as mentioned above, please see: https://github.com/kubeshop/botkube-docs/pull/196/files#diff-09131946afa76d6bdfacbc0ba3fe73c12ffb69e36ef2e2239c62dd5c997a879dR150-R203